packages/test-sync/src/androidTest/kotlin/io/realm/kotlin/test/mongodb/shared/ApiKeyAuthTests.kt

```diff
@@ -0,0 +1,157 @@
+package io.realm.kotlin.test.mongodb.shared
+
+import io.realm.kotlin.internal.platform.runBlocking
+import io.realm.kotlin.mongodb.User
+import io.realm.kotlin.mongodb.auth.ApiKeyAuth
+import io.realm.kotlin.mongodb.exceptions.ServiceException
+import io.realm.kotlin.test.mongodb.TEST_APP_PARTITION
+import io.realm.kotlin.test.mongodb.TestApp
+import io.realm.kotlin.types.ObjectId
+import kotlin.test.AfterTest
+import kotlin.test.BeforeTest
+import kotlin.test.Test
+import kotlin.test.assertEquals
+import kotlin.test.assertFailsWith
+import kotlin.test.assertFalse
+import kotlin.test.assertNotNull
+import kotlin.test.assertNull
+import kotlin.test.assertTrue
+
+class ApiKeyAuthTests {
+    private lateinit var app: TestApp
+    private lateinit var user: User
+    private lateinit var provider: ApiKeyAuth
+    @BeforeTest
+    fun setup() {
+        app = TestApp(appName = TEST_APP_PARTITION)
+        user = app.createUserAndLogin()
+        provider = user.apiKeyAuth
+    }
+
+    @AfterTest
+    fun tearDown() {
+        if (this::app.isInitialized) {
+            app.close()
+        }
+    }
+
+    @Test
+    fun create() = runBlocking {
+        val key = provider.create("foo")
+        assertEquals("foo", key.name)
+        assertNotNull(key.value)
+        assertNotNull(key.id)
+        assertTrue(key.enabled)
+    }
+
+    @Test
+    fun create_throwsWithInvalidName(): Unit = runBlocking {
+        // worth creating a more specific exception?
+        assertFailsWith<ServiceException> {
+            provider.create("%s")
+        }
+    }
+
+    @Test
+    fun create_throwsWithNoName(): Unit = runBlocking {
+        // worth creating a more specific exception?
+        assertFailsWith<ServiceException> {
+            provider.create("")
+        }
+    }
+
```

Comment:
In Realm Java there also exists tests like

```
 @Test
    fun createApiKey_invalidArgumentThrows() {
        testNullArg { provider.create(TestHelper.getNull()) }
        testNullArg { provider.create("") }
        looperThread.runBlocking {
            provider.createAsync(TestHelper.getNull(), checkNullInApiKeyCallback)
        }
        looperThread.runBlocking {
            provider.createAsync("", checkNullInApiKeyCallback)
        }
    }
```

But I'm not entirely sure how this is replicated in kotlin, passing null does not work here due to type safety

Reply:
Yes, these kinds of tests are not relevant in Kotlin since the type system prevents these errors.